lib/tcpip/network_wrapper.cc

```diff
@@ -484,11 +491,75 @@ int network_host_resolve(const char     *hostname,
 	  -1 /* invalid if we are restarting */);
 }
 
+/**
+ * Callback called by FreeRTOS+TCP when a TCP connection is created or
+ * terminated.
+ *
+ * We use this callback to handle the case where a three-way TCP handshake
+ * initiated by a peer on a listening socket fails.
```

Comment:
Assuming this gets called when a socket is ready to be accepted (I think it should but I need to run it and check), this indeed sounds like a good place to plug the futex in.

Reply:
As discussed earlier, this is probably something we want to do as a separate PR, but so that we don't forget it I added a note to https://github.com/CHERIoT-Platform/network-stack/issues/44.